app/uk/gov/hmrc/trackingconsentfrontend/assets/src/common/getLocalBaseUrl.ts

```diff
@@ -0,0 +1,36 @@
+import getCurrentScript from "../common/getCurrentScript";
+
+// IE compatible method for parsing URL
+const extractPortFromUrl = (src: string): string | undefined => {
+    var anchor: HTMLAnchorElement = document.createElement("a");
+    anchor.href = src;
+    const port = anchor.port
+    if (port === '') {
+        return undefined
+    }
+    return port
+}
+
+const getPort = (): string | undefined => {
+    const scriptTag: HTMLScriptElement | undefined = getCurrentScript()
+    if (scriptTag === undefined) {
+        return undefined
+    }
+
+    const src: string | null = scriptTag.getAttribute('src')
+    if (src === null) {
+        return undefined
+    }
+
+    return extractPortFromUrl(src)
+}
+
+const getLocalBaseUrl = (): string => {
```

Comment:
Can you consolidate the full Base URL logic in here?  This does half the job and `getTrackingConsentBaseUrl` does the other half, I'd prefer to see that all in together.

Reply:
Looking at this more closely I'd suggest this:

```
const extractPortFromUrl = (src) => {
    var anchor = document.createElement("a");
    anchor.href = src;
    const host = anchor.host
    const protocol = anchor.protocol
    return [protocol, host].join('//')
}
```

I've tried that in the console feeding in inputs like:

`extractPortFromUrl('https://github.com/abc')`
`extractPortFromUrl('https://github.com:1234/abc')`
`extractPortFromUrl('/abc')`

I think this method is all you need to turn any URL into a baseUrl.

Reply:
The existing logic first tests if you are running locally, if not it returns a *relative* URL. It's only if you are running locally, it then performs the logic to extract the port and construct a localhost URL. This means in the normal case of running in production, the logic is simplest and doesn't involve parsing any URLs, which feels more robust. If we are running locally then it goes to the trouble of parsing the URL, but then we know we are running locally so don't need the full base URL.

Reply:
I'd prefer a simpler function with fewer lines which doesn't have forked logic for local environments.  Can you see a scenario which isn't covered by that approach?